client/scripts/views/components/header/login_selector.tsx

```diff
@@ -0,0 +1,444 @@
+
+    return (
+      <Cui.Menu class="LoginSelectorMenu">
+        <Cui.MenuItem
+          onclick={() =>
+            app.activeChainId()
+              ? navigateToSubpage('/notification-settings')
+              : m.route.set('/notification-settings')
+          }
+          label={
+            <div class="label-wrap">
+              {mobile && m(CustomIcons.CustomBellIcon)}
+              <span>Notification settings</span>
+            </div>
+          }
+        />
+        <Cui.MenuItem
+          onclick={() =>
+            app.activeChainId()
+              ? navigateToSubpage('/settings')
+              : m.route.set('/settings')
+          }
+          label={
+            <div class="label-wrap">
+              {mobile && m(CustomIcons.CustomUserIcon)}
+              <span>Account settings</span>
+            </div>
+          }
+        />
+        <Cui.MenuDivider />
+        <Cui.MenuItem
+          onclick={() => app.modals.create({ modal: FeedbackModal })}
+          label={
+            <div class="label-wrap">
+              {mobile && m(CustomIcons.CustomCommentIcon)}
+              <span>Send feedback</span>
+            </div>
+          }
+        />
+        <Cui.MenuItem
+          onclick={() => {
+            $.get(`${app.serverUrl()}/logout`)
+              .then(async () => {
+                await initAppState();
+                notifySuccess('Logged out');
+                m.redraw();
+              })
+              .catch(() => {
+                // eslint-disable-next-line no-restricted-globals
+                location.reload();
+              });
+            mixpanel.reset();
+          }}
+          label={
+            <div class="label-wrap">
+              {mobile && m(CustomIcons.CustomLogoutIcon)}
+              <span>Logout</span>
+            </div>
+          }
+        />
+      </Cui.Menu>
+    );
+  }
+}
+
+type LoginSelectorAttrs = { small?: boolean };
+
+export class LoginSelector implements m.ClassComponent<LoginSelectorAttrs> {
+  private profileLoadComplete: boolean;
+
+  view(vnode) {
+    const { small } = vnode.attrs;
+
+    if (!app.isLoggedIn())
+      return (
+        <div class="LoginSelector">
+          <div class="login-selector-user">
+            <Cui.Button
+              iconLeft={Cui.Icons.USER}
+              fluid={true}
+              label="Log in"
+              compact={true}
+              size={small ? 'sm' : 'default'}
+              onclick={() => app.modals.create({ modal: LoginModal })}
+            />
+          </div>
+        </div>
+      );
+
+    const activeAddressesWithRole = app.user.activeAccounts.filter(
+      (account) => {
+        return app.user.getRoleInCommunity({
+          account,
+          chain: app.activeChainId(),
+        });
+      }
+    );
+
+    const activeAccountsByRole = app.user.getActiveAccountsByRole();
+
+    const nAccountsWithoutRole = activeAccountsByRole.filter(
+      ([role]) => !role
```

Comment:
This seems like a logic error, the original code was
```
    const nAccountsWithoutRole = activeAccountsByRole.filter(
      ([account, role], index) => !role
    ).length;
```
which destructures and takes the second element, rather than the first.